AccessibilityInsightsForAndroidService/app/src/main/java/com/microsoft/accessibilityinsightsforandroidservice/AccessibilityEventDispatcher.java

```diff
@@ -34,7 +34,9 @@ public AccessibilityEventDispatcher() {
   public void onAccessibilityEvent(AccessibilityEvent event, AccessibilityNodeInfo rootNode) {
     int eventType = event.getEventType();
 
-    if (previousPackageName == null || !previousPackageName.equals(rootNode.getPackageName())) {
+    if (rootNode != null
+        && (previousPackageName == null
+            || !previousPackageName.equals(rootNode.getPackageName()))) {
       previousPackageName = rootNode.getPackageName();
       this.callListeners(onAppChangedListeners, rootNode);
```

Comment:
line 31 `AccessibilityEvent.TYPE_WINDOW_CONTENT_CHANGED,` needs to be removed from the redraw events as it fires every time a highlight is drawn.